src/mono/netcore/System.Private.CoreLib/src/System/GC.Mono.cs

```diff
@@ -248,13 +248,12 @@ internal static ulong GetSegmentSize()
 
         public static GCMemoryInfo GetGCMemoryInfo()
         {
-            _GetGCMemoryInfo(out long highMemoryLoadThresholdBytes,
-                             out long memoryLoadBytes,
-                             out long totalAvailableMemoryBytes,
-                             out long heapSizeBytes,
-                             out long fragmentedBytes);
+            throw new PlatformNotSupportedException();
```

Comment:
It looks like this overload used to work, but it's now being changed to throw?

Reply:
well, because there isn't a mono implementation yet. I did this based on @marek-safar's suggestion but I dunno if I misunderstood him.

Reply:
@Maoni0 I think @marek-safar's suggestion originated from the assumption that this is a new API that we don't yet support and we should just throw exception for now. However, this API turned out to be an extension on the existing `GCMemoryInfo`, and, as Stephen pointed out, we already support part of it and I don't think we should regress what is already implemented. 

We used to just leave fields that we don't yet support uninitialized in the GCMemoryInfo (we might not even support every single field right now) and I remember coreclr was doing the same at some point, when comparing between Windows and Linux, for example. My opinion is that we should leave the fields from the new API with the default value for now and just initialize the fields from the netcore 3.0 version. So instead of throwing, we could just create a new `GCMemoryInfoData` with just these 5 fields initialized and we'll add later support in the mono runtime for the rest of the api.

Reply:
I see; makes sense! does [this](https://github.com/dotnet/runtime/pull/37879/commits/27c06c8b822a07993f07af15c291f21fa1273f16) look good to you? I'm throwing PNSE in the overload but changed the old API the way you suggested.